url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/client/SOCPlayerInterface.java

```diff
@@ -0,0 +1,2292 @@
+            textDisplay.setBounds(x, i.top + 4, w, h);
+            if (! game.isLocal)
+                cdh += 20;
+            chatDisplay.setBounds(x, i.top + 4 + h, w, cdh);
+            h += cdh;
+            textInput.setBounds(x, i.top + 4 + h, w, tfh);
+        } else {
+            // standard size
+            textDisplay.setBounds(i.left + hw + 8, i.top + 4, bw, tdh);
+            chatDisplay.setBounds(i.left + hw + 8, i.top + 4 + tdh, bw, cdh);
+            textInput.setBounds(i.left + hw + 8, i.top + 4 + tah, bw, tfh);
+        }
+        textDisplaysLargerTemp_needsLayout = false;
+
+        npix = textDisplay.getPreferredSize().width;
+        ncols = (int) ((((float) bw) * 100.0) / ((float) npix)) - 2;
+
+        //FontMetrics fm = this.getFontMetrics(textDisplay.getFont());
+        //int nrows = (tdh / fm.getHeight()) - 1;
+
+        //textDisplay.setMaximumLines(nrows);
+        //nrows = (cdh / fm.getHeight()) - 1;
+
+        //chatDisplay.setMaximumLines(nrows);
+        boardPanel.doLayout();
+
+        /**
+         * Reset mouse cursor from WAIT_CURSOR to normal
+         * (set in SOCPlayerClient.startPracticeGame or .guardedActionPerform).
+         */
+        if (layoutNotReadyYet)
+        {
+            client.setCursor(Cursor.getPredefinedCursor(Cursor.DEFAULT_CURSOR));
+            layoutNotReadyYet = false;
+            repaint();
+        }
+    }
+
+    /**
+     * For 6-player board, make the text displays larger/smaller when mouse
+     * enters/exits them.
+     * Timer will set textDisplaysLargerTemp_needsLayout, invalidate(), doLayout().
+     * Wait 200 ms first, to avoid flicker in case of several related
+     * {@link #mouseExited(MouseEvent)}/{@link #mouseEntered(MouseEvent)} events
+     * (such as moving mouse from {@link #textDisplay} to {@link #chatDisplay}).
+     * @since 1.1.08
+     */
+    private void textDisplaysLargerSetResizeTimer()
+    {
+        getEventTimer().schedule(new SOCPITextDisplaysLargerTask(), 200 /* ms */ );
+    }
+
+    /**
+     * Track TextField mouse cursor entry on 6-player board (MouseListener).
+     * Listener is not added unless {@link #is6player}.
+     * For use by {@link #textDisplay}, {@link #chatDisplay}, {@link #textInput}.
+     * Calls {@link #textDisplaysLargerSetResizeTimer()}.
+     * @since 1.1.08
+     */
+    public void mouseEntered(MouseEvent e)
+    {
+        if (! is6player)
+            return;
+
+        final Object src = e.getSource(); 
+        if (src == textDisplay)
+            textDisplayHasMouse = true;
+        else if (src == chatDisplay)
+            chatDisplayHasMouse = true;
+        else if (src == textInput)
+            textInputHasMouse = true;
+        else if (textDisplaysLargerTemp)
+        {
+            if (src == boardPanel)
+                sbFixBHasMouse = true;
+            else if (src == hands[0])
+                sbFixLHasMouse = true;
+            else if (src == hands[1])
+                sbFixRHasMouse = true;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1929:1934, 1970:1975.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.